### PR TITLE
Upgrade to Linter v2

### PR DESCRIPTION
**Pull request by Positive07**
Upgrade to the Standard Linter v2 and Messages v2 protocols used by
Linter v2.

This fixes Issue #25 

I also made sure to update the specs to match the new Message protocol

NOTE: The minimal version of Atom is 1.14 since that is the minimal version required to install Linter v2.0.0

### CODE REVIEW

**Comment by Positive07** [comment]
Do I need to fix that linting issue? It looks like I exceeded the character limit by 1 character in the spec...

File: spec/linter-luacheck-spec.coffee [hunk 1]
```diff
@@ -32,27 +32,27 @@ describe 'The luacheck provider for Linter', ->
...
-          expect(messages[0].text).toEqual("expected identifier near 'local'")
-          expect(messages[0].filePath).toBeDefined()
-          expect(messages[0].filePath).toMatch(/.+error\.lua$/)
-          expect(messages[0].range).toBeDefined()
-          expect(messages[0].range.length).toEqual(2)
-          expect(messages[0].range).toEqual([[2, 6], [2, 11]])
+          expect(messages[0].severity).toBeDefined()
+          expect(messages[0].severity).toEqual('error')
+          expect(messages[0].excerpt).toBeDefined()
+          expect(messages[0].excerpt).toEqual("expected identifier near 'local'")
```
Comment (Arcanemagus) [inline spec/linter-luacheck-spec.coffee]: You could put the expected message in a variable above, then just use the variable here to cut down on the line length.

**Comment by Arcanemagus** [comment]
This linter as a whole needs to be re-written away from COffeeScript, a project for another time though...

**Comment by Positive07** [comment]
I'll fix these asap. Yeah I know what you mean, I'm just trying to keep it alive because it has proven to be really useful.

**Comment by Arcanemagus** [comment]
v1 providers will be supported in linter v2, there is no rush to update the message format 😉.

**Comment by Positive07** [comment]
Is this better?

And yeah I know about v1 being supported, this PR is actually to give it a name, instead of the "Unknown Provider" hahaha... But well the update to v2 was pretty simple so I just made everything in one go!

**Comment by Arcanemagus** [comment]
Published in v1.2.0 🎉.

**Pull request merged**